<launch>    


   <include file="$(find ndt_feature)/launch/replay.launch">
     <arg name="file" value="$(find ndt_feature)/data/SurveyUtanGyro1.bag" />
     <arg name="rate" value="1" />
     <arg name="start_time" value="0" />
   </include>

    <node pkg="ndt_feature" type="ndt_feature2d_fuser" name="ndt_feature2d_fuser" output="screen">

	<param name="laser_topic" value="/laserscan" />
	<param name="matchLaser" value="true" />
	
	<param name="size_x_meters" value="80" />
	<param name="size_y_meters" value="80" />
	<param name="size_z_meters" value="0.2" />
	<param name="resolution" value="0.5" />
	<param name="laser_variance_z" value="0.02" />
	
	<param name="sensor_range" value="29." />
	<param name="min_laser_range" value="0.5" />
	
	<!-- Specific sensor offset parameters for the example file S3000 survey -->
	<param name="sensor_pose_x" value="0.271" />
	<param name="sensor_pose_y" value="-0.001" />
	<param name="sensor_pose_t" value="-0.0049" />

	<!--param name="sensor_pose_x" value="0.230" />
	<param name="sensor_pose_y" value="0.0" />
	<param name="sensor_pose_t" value="0.0021" /-->
	
	<param name="pose_init_x" value="0" />
	<param name="pose_init_y" value="0" />
	<param name="pose_init_t" value="0" />

	<param name="map_directory" value="$(find ndt_fuser)/maps/" />	
	<param name="map_name_prefix" value="office1" />

	<param name="beHMT" value="false" />
	<param name="useOdometry" value="false" />
	<param name="odometry_topic" value="/odom" />
	<param name="visualize" value="false" />
	<param name="plotGTTrack" value="true" />
	<param name="initPoseFromGT" value="true" />
	<param name="gt_topic" value="/state" />
	<param name="tf_odom_frame" value="/state_base_link" />

	<param name="fuser_useNDT" value="true" />
	<param name="fuser_useFeat" value="false" />
	<param name="fuser_useOdom" value="true" />
	<param name="fuser_neighbours" value="0" />
	<param name="fuser_stepcontrol" value="true" />
	<param name="fuser_ITR_MAX" value="100" />
	<param name="fuser_DELTA_SCORE" value="0.0001" />
	<param name="fuser_globalTransf" value="false" />
	<param name="fuser_loadCentroid" value="false" />
	<param name="fuser_forceOdomAsEst" value="false" />
	<param name="fuser_visualizeLocalCloud" value="false" />
	<param name="fuser_allMatchesValid" value="true" />
	<param name="fuser_discardCells" value="false" />
	<param name="fuser_optimizeOnlyYaw" value="false" />
	
	<param name="use_graph" value="true" />

	<param name="graph_newNodeTranslDist" value="4." />
	<param name="occ_map_resolution" value="0.5" />
	<param name="do_pub_occ_map" value="true" />
	<param name="do_pub_debug_markers" value="true" />

	<param name="motion_params_Cd" value="0.005" />
	<param name="motion_params_Ct" value="0.01" />
	<param name="motion_params_Dd" value="0.0001" />
	<param name="motion_params_Cd" value="0.001" />

	<!-- fusion2d bool -->
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ndt_feature)/launch/fuser_laser2d.rviz" />

    <node pkg="tf" type="static_transform_publisher" name="world_map_frame" args="0 0 0 0 0 0 /world /map 10"/>
    <node pkg="tf" type="static_transform_publisher" name="laser_scan_frame" args="0 0 0 0 0 0 /base_laser_link /laserscan 10"/>  
    <node pkg="tf" type="static_transform_publisher" name="world_laser_frame" args="0.230 0.0 0 0 0 0.0021 /fuser /base_laser_link 10"/>  

</launch>
